docs/concepts/CLQL.md

```diff
@@ -253,31 +253,226 @@ Results under include statements appear as children of the parent of the corresp
 A fact with multiple children will match against elements of the code that have child1 *and* child2 *and* child3 etc. The `any_of` operator overrides the implicit "and". The following query finds all String methods that use basic loops:
 
 ```
-method(depth = any):
+common.method(depth = any):
   name: “String”
   any_of:
-    foreach_stmt
-    while_stmt
-    for_stmt
+    common.foreach_stmt
+    common.while_stmt
+    common.for_stmt
 ```
 <!-- TODO(blakemscurr) n_of-->
 
 <br />
 
+# Edge
+
+Facts in AST lexicons refer to nodes in an AST, and the parent/child relationship between facts refers to the parent/child relationship of nodes in the AST. Nodes can have other parent/child relationships that are orthogonal to AST, such as calls. These relationships can be queried with the `edge` keyword.
+
+The following query finds function calls at the top level of a file and follows the `calls` edge to their definition:
+
+```
+common.func_call:
+  edge("calls"):
+    common.func
+```
+
+<br />
+
+# Block
+
+Block statements encapsulate CLQL trees. These subtrees can be repeated with a single argument allowing succint repition of complex patterns. Branches of the subtree can be recombined at the end of the block allowing queries along multiple paths.
+
+## Linear
+
+The following query finds functions with triply nested if statements:
+
+```clql
+common.func:
+  block(repeat = 3):
+    common.if_stmt:
+      blockcontinue
+```
+
+It can be rewritten by expanding the CLQL between `block` and `blockcontinue` 3 times:
```

Comment:
I would swap the order of these. First layout the problem: 

"Say we wanted to find a triple nested if statement, our query would look like the following:"

"With blocks, we can express the same thing like so:"